lib/atlas/graph_from_yaml.rb

```diff
@@ -33,20 +29,35 @@ def build_graph
 
     private
 
-    def set_slots(node, node_attributes)
-      node_attributes.each_pair do |direction, slot_attributes|
-        slot_collection = node.public_send("#{ direction }_slots")
+    def graph_nodes
+      Hash[@graph.nodes.map do |node|
+        attributes       = @graph_yaml.fetch(:nodes)[node.key] || {}
+        node_attributes  = attributes.slice!(:in, :out)
+
+        [ node, { slot: attributes, node: node_attributes } ]
+      end]
+    end
+
+    def update_slots(slots, attributes)
+      attributes.each_pair do |slot_key, share|
+        ref_slot = get_slot(slots, slot_key)
 
-        slot_attributes.each_pair do |slot_key, share_attributes|
-          slot_collection.add(Atlas::Slot.new(
-            node: node,
-            direction: direction,
-            carrier: slot_key
-          ))
+        if share.is_a?(Numeric)
```

Comment:
Why not always export/import both :share and :type? That way the code would be more robust if a situation with both being set ever comes - also the code would be simpler.